.github/workflows/code_quality.yaml

```diff
@@ -0,0 +1,64 @@
+---
+name: Code test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Install rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          component: fmt
+
+      - name: Run fmt checks
+        run: |
+          cargo fmt -- ./lib/src/serviceinfo/models/*.rs
+          cargo fmt -- ./lib/src/tes/models/*.rs
+          cargo fmt -- --check
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Install rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          component: clippy
+
+      - name: Run unit tests
```

Comment:
Run linting?

Reply:
we have format just above that don't we, what do you mean?